specification/sovereign/Sovereign.Management/landingZoneConfigurationResourceProperties.tsp

```diff
@@ -0,0 +1,441 @@
+  Premium: "Premium",
+
+  string,
+}
+
+@doc("Options for resource creation.")
+union ResourceCreationOptions {
+  @doc("Value to create a new resource.")
+  Yes: "Yes",
+
+  @doc("Value to not create a new resource.")
+  No: "No",
+
+  @doc("Value to use an existing resource.")
+  UseExisting: "UseExisting",
+
+  string,
+}
+
+@doc("The request for the create copy action of landing zone configuration.")
+model CreateCopyRequest {
+  @doc("The name of the new landing zone configuration resource.")
+  @pattern("^[a-zA-Z0-9-]{3,24}$")
+  name: string;
+}
+
+@doc("The response of the create copy action of landing zone configuration.")
+model CreateCopyResponse {
+  @doc("The id of the copied landing zone configuration resource.")
+  copiedLandingZoneConfigurationId: armResourceIdentifier<[
+    {
+      type: "Microsoft.Sovereign/landingZoneConfigurations";
+    }
+  ]>;
+}
+
+@doc("The request for the create copy action of landing zone configuration.")
+model CreateLandingZoneConfigurationCopyRequest {
+  @doc("The name of the new landing zone configuration resource.")
+  @pattern("^[a-zA-Z0-9-]{3,64}$")
+  name: string;
+}
+
+@doc("The response of the create copy action of landing zone configuration.")
+model CreateLandingZoneConfigurationCopyResponse {
+  @doc("The id of the copied landing zone configuration resource.")
+  copiedLandingZoneConfigurationId: armResourceIdentifier<[
+    {
+      type: "Microsoft.Sovereign/landingZoneAccounts/landingZoneConfigurations";
+    }
+  ]>;
+}
+
+@doc("The details for the custom resource override.")
+@added(Versions.Preview20250227)
+model CustomNamingConvention {
+  @doc("The type of the resource.")
+  resourceType: ResourceType;
+
+  @doc("The custom formula for the resource type.")
+  formula: string;
+}
+
+@doc("The input values for defining resource types for naming conventions.")
+@added(Versions.Preview20250227)
+union ResourceType {
+  @doc("Value when customer wants to provide custom naming convention for Azure Firewall.")
+  azureFirewalls: "azureFirewalls",
+
+  @doc("Value when customer wants to provide custom naming convention for Log Analytics workspace.")
+  workspaces: "workspaces",
+
+  @doc("Value when customer wants to provide custom naming convention for automation account.")
+  automationAccounts: "automationAccounts",
+
+  @doc("Value when customer wants to provide custom naming convention for dashboard.")
+  dashboards: "dashboards",
+
+  @doc("Value when customer wants to provide custom naming convention for managed identity.")
+  userAssignedIdentities: "userAssignedIdentities",
+
+  @doc("Value when customer wants to provide custom naming convention for Bastion.")
+  bastionHosts: "bastionHosts",
+
+  @doc("Value when customer wants to provide custom naming convention for DDoS protection plan.")
+  ddosProtectionPlans: "ddosProtectionPlans",
+
+  @doc("Value when customer wants to provide custom naming convention for Bastion NSG.")
```

Comment:
Value when customer wants to provide a custom naming convention for Bastion network security group.